src/error.rs

```diff
@@ -72,6 +72,12 @@ pub enum SlateDBError {
 
     #[error("Read channel error: {0}")]
     ReadChannelError(#[from] tokio::sync::oneshot::error::RecvError),
+
+    #[error("background task failed")]
+    BackgroundTaskFailed,
```

Comment:
how about wrap the source error inside it? so we could have more infomative logs when the error occurred

like:

```
    #[error("background task failed: {0}")]
    BackgroundTaskFailed(#[from] Box<dyn Error>)
```

Reply:
I got this to work, but it was ugly 😆 . On a panic the result we get back is `Box<dyn Any + Send>` because panic can be called with any arg that can be sent across threads. It doesn't implement display, and `SlateDbError` needs to be Sync (for the global error register) and Clone (we use this all over the place), so it's:
```
#[error("background task panic'd")]
// we need to wrap the panic args in an Arc so SlateDbError is Clone
// we need to wrap the panic args in a mutex so that SlateDbError is Sync
BackgroundTaskPanic(Arc<Mutex<Box<dyn Any + Send>>>),
```

Reply:
not surprised to see rust's type system being this hard as always 🥲